src/datasets.c

```diff
@@ -1191,12 +1191,12 @@ static int DatasetAddIPv6(Dataset *set, const uint8_t *data, const uint32_t data
         return -1;
     }
 
-    if (data_len != 16) {
+    if (data_len != 16 && data_len != 4) {
```

Comment:
same check in `DatasetAddIPv4` looks odd btw. Shouldn't that be `if (data_len != 4) {`?

Reply:
Yes, it would be better than the < 4. Do you want me to push something on top of the PR ? or I can open an new one.

Reply:
Prefer a new PR

Reply:
Separate from this current one I mean